package/system/procd/patches/0001-ubus-add-bootup-time-measurement.patch

```diff
@@ -0,0 +1,124 @@
+From 6a0d1089a4de97364087405802c53dcd87238947 Mon Sep 17 00:00:00 2001
+From: Florian Eckert <fe@dev.tdt.de>
+Date: Fri, 21 Oct 2022 13:08:38 +0200
+Subject: [PATCH] ubus: add bootup time measurement
+
+Until now it was not possible to determine the booting time. This change
+adds two new elements to the 'ubus system info'.
```

Comment:
I would probably think more about other possible use cases and would try to make it more flexible, instead of just focusing on one specific use case, probably something like the following:

```json
{
      ...
      "init_state": {
          "early" : { "duration": 23.058, "active": false },
          "ubus" : { "duration": 3.52, "active": false },
          "init" : { "duration": 15.22, "active": false },
          "running" : { "duration": 3457.89, "active": true },
          "shutdown": { "duration": 0, "active": false },
      }
}
```
> This flag indicates whether the system is still booting.

So in your use case, you would know, that the system is booting if `running.active` is `false`.